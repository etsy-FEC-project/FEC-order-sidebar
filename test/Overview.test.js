import React from 'react';
import Overview from '../client/src/Overview.jsx';

const testData = {
  materials: ['flesh', 'blood'],
  isHandmade: true,
  isProduct: false,
  whenMade: '90s',
  numReviews: 100,
  numFavorites: 200,
  acceptGiftCards: true,
};

const testNotHandmade = Object.create(testData);
testNotHandmade.isHandmade = false;

describe('Overview', () => {
  it('should display "handmade" text if item is handmade', () => {
    const wrapper = shallow(<Overview data={testData} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not display "handmade" text if item is not handmade', () => {
    const wrapper = shallow(<Overview data={testNotHandmade} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the review and favorite counts, as links', () => {
    const wrapper = shallow(<Overview data={testData} />);
    expect(wrapper.find('a').at(0).text()).toBe('100 reviews');
    expect(wrapper.find('a').at(1).text()).toBe('200 people');
  });

  it('should render the list of materials', () => {
    const wrapper = shallow(<Overview data={testData} />);
    expect(wrapper.find('.overview-list').at(1).text()).toBe('Materials: flesh, blood');
  });

  it('should render the vintage', () => {
    const wrapper = shallow(<Overview data={testData} />);
    expect(wrapper.find('.overview-list').at(2).text()).toBe('Vintage item from the \'90s');
  });
});

