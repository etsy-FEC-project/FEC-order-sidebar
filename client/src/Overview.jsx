import React from 'react';
import PropTypes from 'prop-types';

const Overview = (props) => {
  const renderItemType = () => {
    if (props.data.isHandmade) {
      const text = props.data.isProduct ? 'Handmade item' : 'Handmade supply';
      return <li>{text}</li>;
    }
    // Some more logic here for other item types?
    return null;
  };

  const renderVintage = function renderVintage() {
    if (props.data.whenMade === 'To order') {
      return <li>Made to order</li>;
    }
    if (props.data.whenMade !== 'Recently') {
      return <li>Vintage item from the ${props.data.whenMade}</li>;
    }
    return null;
  };

  const renderGiftCardOption = function renderGiftCardOption() {
    return props.data.acceptGiftCards ?
      <div><span>Img</span><span>This shop accepts Etsy gift cards</span></div> :
      undefined;
  };

  return (
    <div className="mainItem" id="overview">
      <h4>Overview</h4>
      <ul>
        {renderItemType()}
        <li>Materials: {props.data.materials.join(', ')}</li>
        {renderVintage()}
        <li>Feedback: <a href="#">{props.data.numReviews} reviews</a></li>
        <li>Favorited by: <a href="#">{props.data.numFavorites} people</a></li>
      </ul>

      {renderGiftCardOption()}
    </div>
  );
};

Overview.propTypes = {
  data: PropTypes.shape({
    materials: PropTypes.arrayOf(PropTypes.string),
    isHandmade: PropTypes.bool.isRequired,
    isProduct: PropTypes.bool.isRequired,
    whenMade: PropTypes.string.isRequired,
    numReviews: PropTypes.number.isRequired,
    numFavorites: PropTypes.number.isRequired,
    acceptGiftCards: PropTypes.bool.isRequired,
  }).isRequired,
};

export default Overview;
