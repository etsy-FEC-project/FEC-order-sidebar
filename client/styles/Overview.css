@import url('https://fonts.googleapis.com/css?family=Poppins');

.mainItem {
  border-top-style: ridge;
  padding-bottom: 20px;
}

.overviewList {
  font-family: "Poppins", sans-serif;
  white-space: pre;
  font-size: 13px;
  padding-top: 1px;
  color: #444;
}

.overviewTitle {
  font-size: 16px;
  padding-bottom: 3px;
  -webkit-margin-after: 0em;
}

.feedbackLink {
  text-decoration: none;
  color: #E55400;
}

.feedbackLink:hover {
  text-decoration: underline;
  color: #E55400;
}

ul {
  -webkit-margin-before: 0em;
  -webkit-margin-after: 0em;
  -webkit-padding-start: 0px;
  margin-bottom: 10px;
  margin-left: 20px;
}

.giftIcon {
  margin-right: 5px;
}