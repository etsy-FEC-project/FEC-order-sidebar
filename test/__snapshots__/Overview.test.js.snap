// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview should display "handmade" text if item is handmade 1`] = `
<div
  className="main-item"
  id="overview"
>
  <h4>
    Overview
  </h4>
  <ul>
    <li
      className="overview-list"
    >
      Handmade supply
    </li>
    <li
      className="overview-list"
    >
      Materials: 
      flesh, blood
    </li>
    <li
      className="overview-list"
    >
      Vintage item from the 
      '
      90s
    </li>
    <li
      className="overview-list"
    >
      Feedback:
      <a
        href="#"
      >
        100
         reviews
      </a>
    </li>
    <li
      className="overview-list"
    >
      Favorited by:
      <a
        href="#"
      >
        200
         people
      </a>
    </li>
  </ul>
  <div>
    <span
      id="gift-card-icon"
    >
      ((Img))
    </span>
    <span
      id="gift-cart-msg"
    >
      This shop accepts Etsy gift cards
    </span>
  </div>
</div>
`;

exports[`Overview should not display "handmade" text if item is not handmade 1`] = `
<div
  className="main-item"
  id="overview"
>
  <h4>
    Overview
  </h4>
  <ul>
    <li
      className="overview-list"
    >
      Materials: 
      flesh, blood
    </li>
    <li
      className="overview-list"
    >
      Vintage item from the 
      '
      90s
    </li>
    <li
      className="overview-list"
    >
      Feedback:
      <a
        href="#"
      >
        100
         reviews
      </a>
    </li>
    <li
      className="overview-list"
    >
      Favorited by:
      <a
        href="#"
      >
        200
         people
      </a>
    </li>
  </ul>
  <div>
    <span
      id="gift-card-icon"
    >
      ((Img))
    </span>
    <span
      id="gift-cart-msg"
    >
      This shop accepts Etsy gift cards
    </span>
  </div>
</div>
`;
